### PR TITLE
Fix field name generation and accept string identifier for filter entity

### PR DESCRIPTION
**Pull request by alioguz**

### CODE REVIEW

**Comment by alioguz** [comment]
@gilles-g Can you review this pull request please ?

File: Event/Subscriber/DoctrineORMSubscriber.php [hunk 1]
```diff
@@ -86,7 +87,7 @@ public function filterEntity(GetFilterConditionEvent $event)
 
             if ($dqlFrom = $event->getQueryBuilder()->getDQLPart('from')) {
                 $rootPart = reset($dqlFrom);
-                $fieldName = ltrim($event->getField(), $rootPart->getAlias() . '.');
+                $fieldName = \str_replace(\sprintf('%s.', $rootPart->getAlias()), null, $event->getField());
```
Comment (gilles-g) [inline Event/Subscriber/DoctrineORMSubscriber.php:90]: str_replace works better than ltrim in some case?

Comment (alioguz) [reply Event/Subscriber/DoctrineORMSubscriber.php:90]: Using **ltrim** method is not correct here. **ltrim** replace all listing characters and not word.

Example, 2 entities : 

1. **Document** with alias **document**
2. **Category** with alias **category**

```php
dump($event->getField()); // document.category
dump($rootPart->getAlias() . '.'); // document.

// Before (wrong)
dump(ltrim($event->getField(), $rootPart->getAlias() . '.')); // ategory

// After (good)
dump(\str_replace(\sprintf('%s.', $rootPart->getAlias()), null, $event->getField())); // category
```

**Comment by gilles-g** [comment]
Thx @alioguz

**Pull request merged**